### ISSUE TITLE
API: Create error handling component

### ISSUE THREAD

**Issue by pbastia** [bcgov/cas-obps#246]
#### Description:

@DataVillage what would be the responsibility of this component?

Component is responsible for returning the correct error code when an exception is caught. For example, 404 when a resource is not found, 400 when something invalid is posted etc. Further, it provides meaningful error messages with readable timestamps and clear messages.  

Spring Java example at https://www.baeldung.com/global-error-handler-in-a-spring-rest-api.

#### Acceptance Criteria:

Given
Invalid API Request
When
Hitting API endpoint
Then
Return appropriate user friendly error message

#### Development Checklist:

- [x] Create common framework for handling exceptions
- [x] Create handler for 401 (not authorized)
- [x] Create handler for 404 (not found)
- [x] Create hander for 400 (bad request)
- [x] Create custom error handlers as needed
- [x] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ NA] User story is included
- [x] User role and type are identified
- [x] Acceptance criteria are included
- [NA] Wireframes are included (if required)
- [x] Design / Solution is accepted by Product Owner
- [NA] Dependencies are identified (technical, business, regulatory/policy)
- [x] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [x] Acceptance criteria are tested by the CI pipeline
- [NA] UI meets accessibility requirements
- [x] Configuration changes are documented, documentation and designs are updated
- [x] Passes code peer-review
- [x] Passes QA of Acceptance Criteria with verification in Dev and Test
- [x] Ticket is ready to be merged to main branch
- [x] Can be demoed in Sprint Review
- [x] Bugs or future work cards are identified and created
- [x] Reviewed and approved by Product Owner

#### Dependencies

- Blocked by
  API Gateway
- Blocking
  Common, compliance, registration and reporting endpoints

**Comment by DataVillage** [comment]
Component would be responsible for returning the correct error code when an exception is caught.  For example, 404 when a resource is not found, 400 when something invalid is posted etc.  Further, it will provide meaningful error messages with readable timestamps and clear messages.  So instead of a message just saying "bad request" if you post a string when an integer is expected, it will tell you the parameter that has the issue and that the issue is an invalid format (just one example).   

And ideally everything will be performed using a common error handler.  Here is a Spring Java example https://www.baeldung.com/global-error-handler-in-a-spring-rest-api.  Ideally, something similar can be done with Django.

**Comment by pbastia** [comment]
I believe @BCerki 's [latest PR](https://github.com/bcgov/cas-registration/pull/1259) creates that component. We might be able to close this
